Add newsletter prompt and link

AnimeAllstar opened the pull request:
closes #246

kewbish commented:
Code looks good so far, thank you!

There's currently a bug where if you click 'maybe later' on the modal, then go to another page, the modal will pop up again (for example, I click 'maybe later' on the homepage but go to the volunteering page and see the modal again). I think [setting the `path=/` on the cookie](https://stackoverflow.com/a/18155490) should make it so that the same cookie's accessible from the whole site? 

Some other thoughts:
- Should we consider making the modal closable via clicking outside it (not setting `data-bs-backdrop` to static)? Currently, it feels a bit forced and might drive users away from our site.
- ^ would make the modal consistent with keyboard behaviour too. Currently, if you press ESC, the modal closes, but without setting the cookie properly. I think using something like [this](https://stackoverflow.com/questions/12319171/how-to-handle-the-modal-closing-event-in-twitter-bootstrap) and binding an event listener to the close event of the modal would cover all the edge cases.
- Should we put a `setInterval` on the modal showing to delay it by a couple seconds? On other sites or forums there's usually something like this so the user can look through content for a bit before we prompt them.

Just ideas though, what do you think?

AnimeAllstar commented:
- Removed data-bs-backdrop `static`
- Added path=/ to cookies
- Added a 10-second delay in showing the newsletter popup
- Added handling for ESC key and clicking outside modal using `hide.bs.modal`

kewbish reviewed (approved):
Very nice, thanks :tada:
